src/python/pants/init/engine_initializer.py

```diff
@@ -200,15 +201,17 @@ def run_console_rules(self, goals, target_roots):
     # Reduce to only applicable goals - with validation happening by way of `validate_goals()`.
     goals = [goal for goal in goals if goal in self.goal_map]
     subjects = self._determine_subjects(target_roots)
+    console = Console()
     # Console rule can only have one subject.
     assert len(subjects) == 1
     for goal in goals:
       try:
         goal_product = self.goal_map[goal]
+        params = Params(subjects[0], options_bootstrapper, console)
```

Comment:
This is my ignorance of the new engine talking, but where are these params consumed? I.e., what is using the options_bootstrapper in the console rule?

Reply:
When the engine is invoked with a request for some type (a synthetic type `ListGoalExecution` in this case), the engine selects a root `@rule` to execute that produces that type given a non-ambiguous [subset of the input params](https://github.com/pantsbuild/pants/blob/master/src/rust/engine/src/rule_graph.rs#L917-L970).

In this case, `ListGoalExecution` actually uses all 3 of these parameters to execute: a portion of the rendered rulegraph looks like:
![list-goal-execution-portion](https://user-images.githubusercontent.com/46740/50658619-c9dbc600-0f4e-11e9-9594-bee05f5b49c7.png)

Reply:
That much I understood already :) What I'm missing is where in the code the bootstrapper param is used in the production of the type.

Reply:
@benjyw : See the screenshot. It shows that:
* `fast_list` depends on
* `construct_scope_fastlist` (a synthetic method generated by this patch), which depends on
* `scope_options`, which depends on
* `parse_options` which consumes the `OptionsBootstrapper`.